backend/src/main/kotlin/code/nebula/cipherquest/service/UserLevelService.kt

```diff
@@ -115,7 +118,14 @@ class UserLevelService(
         request: CreateUserLevelRequest,
         storyName: String,
     ): UserLevel {
-        val username = request.email.substringBefore("@")
+        val baseUsername = request.email.substringBefore("@")
+
+        val username =
+            generateSequence(
+                baseUsername,
+            ) { "$baseUsername${Random.nextInt(LOWER_LIMIT_USERNAME_RANDOM_ID, UPPER_LIMIT_USERNAME_RANDOM_ID)}" }
```

Comment:
Using `Random.nextInt(until: Int)` we can omit `from`, as it is already equal to 0.
```suggestion
            ) { "$baseUsername${Random.nextInt(UPPER_LIMIT_USERNAME_RANDOM_ID)}" }
```